st2common/st2common/models/db/auth.py

```diff
@@ -49,4 +53,32 @@ class TokenDB(stormbase.StormFoundationDB):
                             help_text='Arbitrary metadata associated with this token')
 
 
-MODELS = [UserDB, TokenDB]
+class ApiKeyDB(stormbase.StormFoundationDB, stormbase.UIDFieldMixin):
+    """
+    """
+    RESOURCE_TYPE = ResourceType.API_KEY
+    UID_FIELDS = ['id']
+
+    user = me.StringField(required=True)
+    key_hash = me.StringField(required=True, unique=True)
+    metadata = me.DictField(required=False,
```

Comment:
Edit: I just saw how you used metadata in the fixtures / examples below. That's fine we should just document that and encourage users to populate this field :)